etcdmain/config.go

```diff
@@ -177,6 +178,7 @@ func NewConfig() *config {
 		// Should never happen.
 		plog.Panicf("unexpected error setting up clusterStateFlag: %v", err)
 	}
+	fs.BoolVar(&cfg.strictReconfigCheck, "strict-reconfig-check", false, "Reject reconfiguration if it can cause potential deadlock")
```

Comment:
I guess we need to improve the message "Reject reconfiguration if it can cause potential deadlock".

Potential deadlock is not that obvious to a lot users. 

"Reject reconfiguration that might cause quorum loss"